source/external/cJSON.cpp

```diff
@@ -426,7 +426,7 @@ CJSON_PUBLIC(char*) cJSON_SetValuestring(cJSON *object, const char *valuestring)
     }
     if (strlen(valuestring) <= strlen(object->valuestring))
     {
-        strncpy(object->valuestring, valuestring, strlen(valuestring) + 1);
```

Comment:
This causes 
```
error: char* strncpy(char*, const char*, size_t) specified bound depends on the length of the source argument [-Werror=stringop-truncation]
  429 |         strncpy(object->valuestring, valuestring, strlen(valuestring) + 1);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/aws-crt-cpp/source/external/cJSON.cpp:427:15: note: length computed here
  427 |     if (strlen(valuestring) <= strlen(object->valuestring))
  ```

Reply:
Question: Should this be `size_t value_length = strlen(valuestring) + sizeof("");` like in aws-c-common: https://github.com/awslabs/aws-c-common/blob/3973e3c89bbb532f5c3d2327b602836a57bd4db6/source/external/cJSON.c#L410?

Reply:
I think `sizeof("")` is always 1. So it doesn't make a difference. https://stackoverflow.com/questions/4141666/why-sizeof-is-equivalent-to-1-and-sizeofnull-is-equivalent-to-4-in-c-langu

Reply:
> Not related, but should we remove the copy of cJSON from aws-crt-cpp, as we have one in aws-c-common already?

I am working on this now! It pretty much maps 1:1 with what we have in `aws-c-common`.